app/Models/Collection.php

```diff
@@ -24,6 +25,18 @@ public function collectionType()
     return $this->belongsTo(CollectionType::class);
   }
 
+  public static function findArchivalIdentifier($collectionId)
```

Comment:
Maybe cut this and just check the value of the archival_identifier in an instantiated collection? If null is never allowed, define this as a validator and/or 'NOT NULL' in the migration.

Reply:
Got it. I already have it as validation on the request so it should be good to go.